File: graph.py
import networkx as nx
import matplotlib.pyplot as plt
import json

# Load your graph data
with open("sample.json") as f:
    data = json.load(f)

G = nx.Graph()
for edge in data["edges"]:
    G.add_edge(edge["u"], edge["v"], weight=edge["w"])

pos = nx.spring_layout(G)  # auto layout like GPS
nx.draw(G, pos, with_labels=True, node_color="lightblue", node_size=700)
labels = nx.get_edge_attributes(G, "weight")
nx.draw_networkx_edge_labels(G, pos, edge_labels=labels)

plt.show()
 

File: graph.cpp
#include "graph.h"

void Graph::add_edge(const std::string& from, const std::string& to, double weight) {
    adj[from].push_back({to, weight});
}


std::unordered_map<std::string, std::vector<Edge>>& Graph::get_adjacency_list() {
    return adj;
}